Core.DomainModel/GDPR/DataProcessingRegistration.cs

```diff
@@ -340,6 +340,48 @@ public Maybe<IEnumerable<DataProcessingRegistrationOversightDate>> SetOversightC
         }
 
         public virtual ICollection<ItContract.ItContract> AssociatedContracts { get; set; }
+        public int? MainContractId { get; set; }
+        public virtual ItContract.ItContract MainContract { get; set; }
+        public bool IsActiveAccordingToMainContract => CheckContractValidity();
+
+        public void ResetMainContract()
+        {
+            MainContract?.Track();
+            MainContract = null;
+        }
+
+        public Maybe<OperationError> AssignMainContract(int contractId)
+        {
+            if (MainContract != null && contractId == MainContract.Id)
+                return Maybe<OperationError>.None;
+
+            var contract = AssociatedContracts.FirstOrDefault(c => c.Id == contractId);
+            if (contract == null)
+                return new OperationError($"Contract with id: {contractId} is not associated with this data processing registration", OperationFailure.BadState);
+
+            ResetMainContract();
+            MainContract = contract;
+
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> RemoveMainContract(int contractId)
+        {
+            if (MainContract == null)
+                return Maybe<OperationError>.None;
+
+            if(MainContract.Id != contractId)
+                return new OperationError($"Contract with id: {contractId} is not set as the main contract for this data processing registration", OperationFailure.BadInput);
+
+            ResetMainContract();
+
+            return Maybe<OperationError>.None;
+        }
+
+        private bool CheckContractValidity()
+        {
+            return MainContract?.IsActive != false;
```

Comment:
might be a good idea with a comment in the code stating that no contract also mean valid. A person without that knowledge might be tempted to change this expression to == true, but that would actually not return the right value. You could also make it much more explitic:

`return MainContract.FromNullable().Select(c=>c.IsActive).GetValueOrFallback(true);`

or
```
return MainContract == null || MainContract.IsActive == true;
```